lua/gitlab/reviewer/diffview.lua

```diff
@@ -120,7 +128,11 @@ M.get_location = function(range)
   local result = {}
   local type
   local is_new
-  if layout.a.file.bufnr == bufnr then
+
+  if
+    layout.a.file.bufnr == bufnr
+    or (M.lines_are_same(view.cur_layout) and layout.b.file.bufnr == bufnr and range == nil)
+  then
```

Comment:
If leaving a comment in the "b" (aka new) pane, make sure that the line is not identical with the old line. If it is, then we are actually leaving a comment on the old diff.